cuda_core/cuda/core/experimental/_device.py

```diff
@@ -62,7 +62,13 @@ def __new__(cls, device_id=None):
                 for dev_id in range(total):
                     dev = super().__new__(cls)
                     dev._id = dev_id
-                    dev._mr = _DefaultAsyncMempool(dev_id)
+                    # If the device is in TCC mode, or does not support memory pools for some other reason,
+                    # use the SynchronousMemoryResource which does not use memory pools.
+                    if (handle_return(cudart.cudaGetDeviceProperties(dev_id))).memoryPoolsSupported == 1:
```

Comment:
nit: could you switch to use `cudaDeviceGetAttribute` and check for `cudaDevAttrMemoryPoolsSupported` (or the driver-equivalent `cuDeviceGetAttribute` + `CU_DEVICE_ATTRIBUTE_MEMORY_POOLS_SUPPORTED`)? It should be faster than getting all properties back at once, which would help reduce `Device` initialization time.